ir/src/main/scala/org/scalajs/core/ir/Utils.scala

```diff
@@ -66,33 +67,39 @@ object Utils {
     // scalastyle:on return
   }
 
-  private def createEscapeJSString(str: String): String = {
+  def printEscapeJS(str: String, out: java.io.Writer): Unit = {
     val end = str.length()
-    val builder = new java.lang.StringBuilder
-    builder.ensureCapacity(end * 2)
     var i = 0
     while (i != end) {
       val c = str.charAt(i)
       if (c >= 32 && c <= 126 && c != '\\' && c != '"') {
-        builder.append(c) // ASCII printable characters
+        out.write(c) // ASCII printable characters
       } else {
-        def encode(c: Char): String = f"\\u$c%04x"
-        builder.append((c: @switch) match {
-          case '\\'     => "\\\\"
-          case '"'      => "\\\""
-          case '\u0007' => "\\a"
-          case '\u0008' => "\\b"
-          case '\u0009' => "\\t"
-          case '\u000A' => "\\n"
-          case '\u000B' => "\\v"
-          case '\u000C' => "\\f"
-          case '\u000D' => "\\r"
-          case c        => encode(c)
-        })
+        def escapeJSEncoded(c: Char): Unit = {
+          out.write('\\')
+          (c: @switch) match {
+            case '\u0007' => out.write('a')
+            case '\u0008' => out.write('b')
+            case '\u0009' => out.write('t')
+            case '\u000A' => out.write('n')
+            case '\u000B' => out.write('v')
+            case '\u000C' => out.write('f')
+            case '\u000D' => out.write('r')
+            case '"'      => out.write('\"')
+            case '\\'     => out.write('\\')
+            case c        => out.write(f"u$c%04x")
+          }
+      }
+        escapeJSEncoded(c)
       }
       i += 1
     }
-    builder.toString()
+  }
+
+  private def createEscapeJSString(str: String): String = {
```

Comment:
This method should be before `printEscapeJS`, since it is only a helper for `escapeJS`.